src/screens/OtherProfile/elements/BioAndChat.js

```diff
@@ -76,15 +89,13 @@ const BioAndChat = (props) => {
       };
       const response = await sendAnonymousDMOtherProfile(anonDMParams);
       await saveChatFromOtherProfile(response, 'sent', true, ANON_PM);
-      setDmChat('');
     } catch (e) {
       if (e?.response?.data?.status === CHANNEL_BLOCKED) {
         const response = e?.response?.data?.data;
-        await savePendingChatFromOtherProfile(response, true);
-        setDmChat('');
+        await savePendingChatFromOtherProfile(response, 'pending', true, ANON_PM);
         return;
       }
-    } finally {
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the error handling here could be improved. If the channel is blocked, the function returns early and `setLoadingSendDM(false);` is not called, which could leave the loading state in an inconsistent state. Consider moving `setLoadingSendDM(false);` to a finally block to ensure it always gets called.

```diff
  } catch (e) {
    if (e?.response?.data?.status === CHANNEL_BLOCKED) {
      const response = e?.response?.data?.data;
      await savePendingChatFromOtherProfile(response, 'pending', true, ANON_PM);
      return;
    }
-   setLoadingSendDM(false);
  }
+ finally {
+   setLoadingSendDM(false);
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->